Source/MapComponents/MapComponent_RenderManager.cs

```diff
@@ -476,16 +477,46 @@ private void SaveUnityEncoding(byte[] encodedImage)
             var filePath = CreateCurrentFilePath();
 
             File.WriteAllBytes(filePath, encodedImage);
+
             // Create tmp copy to file if needed
             if (!manuallyTriggered && PRModSettings.fileNamePattern == FileNamePattern.BothTmpCopy)
             {
                 File.Copy(filePath, CreateFilePath(FileNamePattern.Numbered, true));
             }
-
+            if (PRModSettings.encoding == EncodingType.UnityJPG & PRModSettings.qualityAdjustment == JPGQualityAdjustmentSetting.Automatic)
+            {
+                AdjustJPGQuality(filePath);
+            }
             DoEncodingPost();
         }
 
-        private string CreateCurrentFilePath()
+        private void AdjustJPGQuality(string filePath)
+        {
+            // Adjust JPG quality to reach target filesize
+            if (File.Exists(filePath))
+            {  
+                FileInfo RenderInfo = new FileInfo(filePath);
+                long RenderLenght = RenderInfo.Length / 1048576;   
+                if (RenderLenght > PRModSettings.renderSize)
+                {
+                    if (PRModSettings.JPGQuality > 0)
+                    {
+                        PRModSettings.JPGQuality -= 1;
+                        Messages.Message("JPG quality decreased to " + PRModSettings.JPGQuality.ToString() + "% · Render size: " + RenderLenght.ToString() + " Target: " + PRModSettings.renderSize.ToString(), MessageTypeDefOf.CautionInput, false);
+                    }
+                }
+                else if (RenderLenght < PRModSettings.renderSize)
```

Comment:
```suggestion
                else if (RenderLenght <= PRModSettings.renderSize)
```

Reply:
Why change quality if it's satisfactory? That will cause bouncing, I think

Reply:
Ah, you're right.